packages/forms/frontend/sirius-components-forms/src/propertysections/SplitButtonPropertySection.tsx

```diff
@@ -0,0 +1,322 @@
+import { getTextDecorationLineValue } from './getTextDecorationLineValue';
+
+const useStyle = makeStyles<Theme, ButtonStyleProps>((theme) => ({
+  style: {
+    backgroundColor: ({ backgroundColor }) =>
+      backgroundColor ? getCSSColor(backgroundColor, theme) : theme.palette.primary.light,
+    color: ({ foregroundColor }) => (foregroundColor ? getCSSColor(foregroundColor, theme) : 'white'),
+    fontSize: ({ fontSize }) => (fontSize ? fontSize : null),
+    fontStyle: ({ italic }) => (italic ? 'italic' : null),
+    fontWeight: ({ bold }) => (bold ? 'bold' : null),
+    textDecorationLine: ({ underline, strikeThrough }) => getTextDecorationLineValue(underline, strikeThrough),
+    '&:hover': {
+      backgroundColor: ({ backgroundColor }) =>
+        backgroundColor ? getCSSColor(backgroundColor, theme) : theme.palette.primary.main,
+      color: ({ foregroundColor }) => (foregroundColor ? getCSSColor(foregroundColor, theme) : 'white'),
+      fontSize: ({ fontSize }) => (fontSize ? fontSize : null),
+      fontStyle: ({ italic }) => (italic ? 'italic' : null),
+      fontWeight: ({ bold }) => (bold ? 'bold' : null),
+      textDecorationLine: ({ underline, strikeThrough }) => getTextDecorationLineValue(underline, strikeThrough),
+    },
+    '&.Mui-disabled': {
+      color: theme.palette.action.disabled,
+      backgroundColor: theme.palette.action.disabledBackground,
+      opacity: 1,
+    },
+  },
+  icon: {
+    marginRight: ({ iconOnly }) => (iconOnly ? theme.spacing(0) : theme.spacing(2)),
+    height: theme.spacing(2),
+    width: theme.spacing(2),
+  },
+}));
+
+export const updateWidgetFocusMutation = gql`
+  mutation updateWidgetFocus($input: UpdateWidgetFocusInput!) {
+    updateWidgetFocus(input: $input) {
+      __typename
+      ... on ErrorPayload {
+        messages {
+          body
+          level
+        }
+      }
+    }
+  }
+`;
+
+export const pushButtonMutation = gql`
+  mutation pushButton($input: PushButtonInput!) {
+    pushButton(input: $input) {
+      __typename
+      ... on ErrorPayload {
+        messages {
+          body
+          level
+        }
+      }
+      ... on SuccessPayload {
+        messages {
+          body
+          level
+        }
+      }
+    }
+  }
+`;
+
+const isErrorPayload = (payload: GQLPushButtonPayload | GQLUpdateWidgetFocusPayload): payload is GQLErrorPayload =>
+  payload.__typename === 'ErrorPayload';
+
+const isSuccessPayload = (payload: GQLPushButtonPayload | GQLUpdateWidgetFocusPayload): payload is GQLSuccessPayload =>
+  payload.__typename === 'SuccessPayload';
+
+export const SplitButtonPropertySection = ({
+  editingContextId,
+  formId,
+  widget,
+  subscribers,
+  readOnly,
+}: SplitButtonPropertySectionProps) => {
+  const { addErrorMessage, addMessages } = useMultiToast();
+  const [state, setState] = useState<SplitButtonState>({
+    open: false,
+    selectedIndex: 0,
+  });
+
+  const anchorRef = React.useRef<any>(null);
```

Comment:
Please set the real type of the `useRef `(instead of any) and remove `React.` part.